systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaNodePoolResource.java

```diff
@@ -118,7 +124,9 @@ public static KafkaNodePool convertKafkaResourceToKafkaNodePool(Kafka resource)
 
     public static LabelSelector getLabelSelector(String clusterName, String poolName, ProcessRoles processRole) {
         Map<String, String> matchLabels = new HashMap<>();
-        matchLabels.put(Labels.STRIMZI_CLUSTER_LABEL, clusterName);
+        if (clusterName != null) {
+            matchLabels.put(Labels.STRIMZI_CLUSTER_LABEL, clusterName);
+        }
```

Comment:
Just out of curiosity - in which cases it will be null? I mean, the clusterName should be filled or?

Reply:
> Just out of curiosity - in which cases it will be null? I mean, the clusterName should be filled or?
Check the next overloaded method :D, 
```java
public static LabelSelector getLabelSelector(final String poolName, final ProcessRoles processRoles) {
        return getLabelSelector(null, poolName, processRoles);
    }
```
Well, clusterName could be retrieved from KNP labels but it is not mandatory to do so. When you see that I am getting this LabelSelector only using the KNP name + role
```
@Override
    public boolean waitForReadiness(KafkaNodePool resource) {
        final LabelSelector kafkaNodePoolSelector = KafkaNodePoolResource.getLabelSelector(resource.getMetadata().getName(), resource.getSpec().getRoles().get(0));

        // here we wait for readiness of the Kafka pods related to KafkaNodePools
        PodUtils.waitForPodsReady(resource.getMetadata().getNamespace(), kafkaNodePoolSelector, resource.getSpec().getReplicas(), true);

        return true;
    }
```